OptimizelySDKCore/OptimizelySDKCore/OPTLYNotificationCenter.m

```diff
@@ -91,12 +86,12 @@ - (void)clearAllNotifications {
 
 - (void)sendNotifications:(OPTLYNotificationType)type args:(NSArray *)args {
     OPTLYNotificationHolder *notification = _notifications[@(type)];
-    for (id<OPTLYNotificationDelegate> object in notification.allValues) {
+    for (GenericListener object in notification.allValues) {
         @try {
             if (type == OPTLYNotificationTypeActivate)
-                [self invokeSelector:object selector:@selector(onActivate:userId:attributes:variation:event:) arguments:args];
+                [self notifyActivateListener:((ActivateListener) object) args:args];
             else
```

Comment:
one more change here.  since we support GenericListeners and adding new listener types (via category or extension), we should probably have case statement such as the following

switch (type)
 case OPTLYNotificationTypeActivate:
    [self notifyActivateListener...]
    break;
 case OPTLYNotificationTypeTrack:
   [self notifyTrackListener...];
   break;
 default:
   (GenericListener) object(args);
}

Right now, if it isn't Activate you call track without testing the type.

Reply:
while we're at it, let's change object to listener.